runtime/sema/checker.go

```diff
@@ -1559,16 +1559,19 @@ func (checker *Checker) recordResourceInvalidation(
 	switch expression.(type) {
 	case *ast.MemberExpression:
 
-		if accessedSelfMember == nil ||
-			!checker.allowSelfResourceFieldInvalidation {
+		if (accessedSelfMember == nil ||
+			!checker.allowSelfResourceFieldInvalidation) &&
+			invalidationKind != ResourceInvalidationKindMoveTemporary {
```

Comment:
Curious why this additional check was needed? is there any particular test that is failing without it?

Reply:
Yeah, if you remove this then the checking for `let x <- r.y <- e` will fail. Normally a field member cannot be invalidated in this way, which is what this check is designed to prevent. If we don't add this case, then attempting to invalidate the middle `e1` in `let x <- e1 <- e2` will fail, since in this example `e1` is a member access and can't be invalidated. However, it's okay in this case because the invalidation is temporary (hence the `Temporary` move kind), so I just added an extra condition here to not report an erroneous error for this specific case.